pkg/thanos/statefulset.go

```diff
@@ -147,7 +147,11 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 	thanosVersion := operator.StringValOrDefault(tr.Spec.Version, operator.DefaultThanosVersion)
 	if _, err := semver.ParseTolerant(thanosVersion); err != nil {
 		return nil, fmt.Errorf("failed to parse Thanos version: %w", err)
+	}
 
+	version, err := semver.ParseTolerant(thanosVersion)
```

Comment:
we already call `semver.ParseTolerant(thanosVersion)` above. Can you rearrange to have only one call?

Reply:
oh sorry I didn't see it